qst/type/src/main/java/io/deephaven/qst/type/PrimitiveVectorType.java

```diff
@@ -73,10 +73,55 @@ public final <R> R walk(ArrayType.Visitor<R> visitor) {
     }
 
     @Check
-    final void checkClazz() {
-        if (!VALID_CLASSES.contains(clazz().getName())) {
-            throw new IllegalArgumentException(String.format("Class '%s' is not a valid '%s'",
-                    clazz(), PrimitiveVectorType.class));
+    final void checkPairing() {
+        final String vectorClassNameFromComponent = componentType().walk(VectorClassName.INSTANCE);
+        if (!clazz().getName().equals(vectorClassNameFromComponent)) {
+            throw new IllegalArgumentException(String.format("Invalid PrimitiveVectorType. clazz=%s, componentType=%s",
+                    clazz().getName(), componentType()));
+        }
+    }
+
+    private enum VectorClassName implements PrimitiveType.Visitor<String> {
+        INSTANCE;
+
+        @Override
+        public String visit(BooleanType booleanType) {
+            return null;
```

Comment:
Is it possible (or necessary?) to map this to `ObjectVector<Boolean>` -- this is what a groupBy will do for both boxed and unboxed `boolean`s.

Reply:
So, this is mostly a modelling Q;  `ObjectVector<Integer>` is not the same as `IntVector`. In that vain, `ObjectVector<Boolean>` is a valid `io.deephaven.qst.type.GenericVectorType`, but there is no `io.deephaven.qst.type.PrimitiveVectorType` equivalent (just as there is no `BoolVector`).